server/src/main/java/org/opensearch/cluster/coordination/Coordinator.java

```diff
@@ -865,7 +866,14 @@ protected void doStart() {
 
     @Override
     public DiscoveryStats stats() {
-        return new DiscoveryStats(new PendingClusterStateStats(0, 0, 0), publicationHandler.stats());
+        CoordinationState.PersistedState remotePersistedState = persistedStateRegistry.getPersistedState(
+            PersistedStateRegistry.PersistedStateType.REMOTE
+        );
+        ClusterStateStats clusterStateStats = clusterManagerService.getStateStats();
+        if (remotePersistedState != null) {
+            clusterStateStats.setRemoteStateStats(remotePersistedState.getPersistedStateStats());
+        }
```

Comment:
you can check if stats are available for any of the persisted state and add them to an array
```
ArrayList<Long> stats = new ArrayList();
        Stream.of(PersistedStateRegistry.PersistedStateType.values()).forEach(val -> {
            stats.add(persistedStateRegistry.getPersistedState(val).getStats()); });
```

Reply:
Using an array now to auto add new stats when we add in future.